### DISCUSSION TITLE
Gene type

### DISCUSSION THREAD

**Discussion by soheilminer** [ahmedfgad/GeneticAlgorithmPython#155] [General]
Hello everyone, 

I'm using this package for my research in stochastic mine planning, and I'm trying to adapt it to my own problem.
I'm just wondering if I can define each gene as a tuple that contains two values, and only perform the crossover/mutation functions on the second value while the first one is fixed. Any help is appreciated.

**Comment by ahmedfgad** [comment]
Hello,

Yes this is possible using the `gene_space` parameter. This is an example which makes the value of the first gene fixed (1) and the value of the second one is `None` so that it can be randomly changed using the GA.

```python
ga_instance = pygad.GA(...,
                       gene_space=[[1], None],
                       ...)
```
